services/src/main/scala/cromwell/services/metadata/impl/pubsub/PubSubMetadataServiceActor.scala

```diff
@@ -0,0 +1,133 @@
+package cromwell.services.metadata.impl.pubsub
+
+import akka.actor.{Actor, ActorLogging, Props}
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.config.Config
+import cromwell.cloudsupport.gcp.GoogleConfiguration
+import cromwell.cloudsupport.gcp.auth.ServiceAccountMode
+import cromwell.cloudsupport.gcp.auth.ServiceAccountMode.PemFileFormat
+import cromwell.core.Dispatcher._
+import cromwell.services.metadata._
+import cromwell.services.metadata.MetadataService.{MetadataWriteFailure, MetadataWriteSuccess, PutMetadataAction, PutMetadataActionAndRespond}
+import net.ceedubs.ficus.Ficus._
+import org.broadinstitute.dsde.workbench.google.{GooglePubSubDAO, HttpGooglePubSubDAO}
+import spray.json._
+import cats.instances.future._
+import cats.syntax.functor._
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+/**
+  * A *write-only* metadata service implementation which pushes all events to Google PubSub. The expectation is that
+  * metadata reads are being handled outside of this Cromwell instance.
+  *
+  * For now it requires a google service account auth with a PEM file. If the underlying PubSub library is updated to
+  * allow for other forms of auth this becomes viable for us. Note that currently Cromwell complains that PEM is
+  * deprecated, so we might need to contribute a patch to workbench-lib's pubsub tooling.
+  *
+  * This is intentionally not documented, considering there exists no way to consume/read metadata if this is active
+  * anyone using this is by definition more advanced than the Cromwell developers
+  *
+  * Potential TODOs to consider if moving this forward from proof-of-concept:
+  *  - This will probably need some measure of batching, need to see it in action before proceeding down that path
```

Comment:
Doesn't Google's client have batching?

Reply:
This is using the client built in `workbench-lib` which has batching. Batching isn't cut & dry in pubsub and has some nuance. This comment is saying that for now that nuance is being punted on.